packages/transaction-controller/src/utils/resimulate.ts

```diff
@@ -0,0 +1,288 @@
+import type { Hex } from '@metamask/utils';
+import { createModuleLogger, remove0x } from '@metamask/utils';
+import { BN } from 'bn.js';
+import { isEqual } from 'lodash';
+
+import { projectLogger } from '../logger';
+import type {
+  SimulationBalanceChange,
+  SimulationData,
+  TransactionMeta,
+  TransactionParams,
+} from '../types';
+import { getPercentageChange } from './utils';
+
+const log = createModuleLogger(projectLogger, 'resimulate');
+
+export const RESIMULATE_PARAMS = ['to', 'value', 'data'] as const;
+export const BLOCKAID_RESULT_TYPE_MALICIOUS = 'Malicious';
+export const VALUE_COMPARISON_PERCENT_THRESHOLD = 5;
+export const BLOCK_TIME_ADDITIONAL_SECONDS = 60;
+
+export type ResimulateResponse = {
+  blockTime?: number;
+  resimulate: boolean;
+};
+
+/**
+ * Determine if a transaction should be resimulated.
+ * @param originalTransactionMeta - The original transaction metadata.
+ * @param newTransactionMeta - The new transaction metadata.
+ * @returns Whether the transaction should be resimulated.
+ */
+export function shouldResimulate(
+  originalTransactionMeta: TransactionMeta,
+  newTransactionMeta: TransactionMeta,
+) {
+  const { id: transactionId } = newTransactionMeta;
+
+  const parametersUpdated = isParametersUpdated(
+    originalTransactionMeta,
+    newTransactionMeta,
+  );
+
+  const securityAlert = hasNewSecurityAlert(
```

Comment:
[nit] for the boolean variables here, it's suggested that we prepend the names with verbs for improved readability which help mitigate misinterpretations 

ref:
- https://github.com/MetaMask/contributor-docs/blob/main/docs/javascript.md#use-a-verb-phrase-for-boolean-variable-names-that-do-not-involve-secondary-objects
- [TypeScript ESLint](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-that-boolean-variables-are-prefixed-with-an-allowed-verb)
 "prefix": ["is", "should", "has", "can", "did", "will"]

Reply:
This is intentional to not clash with the function names. Also low risk here since the values aren't subsequently returned.